test_scipy_like fails with meson 0.63.3: backport landed in 0.63.3

On openSUSE Tumbleweed, an update from meson 0.63.2 to 0.63.3 produces a new failure in the test suite:

```
[    5s] + PYTHONPATH=/home/abuild/rpmbuild/BUILDROOT/python-meson-python-0.10.0-0.x86_64/usr/lib/python3.8/site-packages
[    5s] + PYTHONDONTWRITEBYTECODE=1
[    5s] + pytest-3.8 --ignore=_build.python38 --ignore=_build.python39 --ignore=_build.python310 -v -vv
[    5s] ============================= test session starts ==============================
[    5s] platform linux -- Python 3.8.14, pytest-7.1.2, pluggy-1.0.0 -- /usr/bin/python3.8
[    5s] cachedir: .pytest_cache
[    5s] rootdir: /home/abuild/rpmbuild/BUILD/meson_python-0.10.0, configfile: setup.cfg
[    5s] plugins: mock-3.6.1
[    6s] collecting ... collected 53 items
[    6s] 
....
[   24s] =================================== FAILURES ===================================
[   24s] _______________________________ test_scipy_like ________________________________
[   24s] 
[   24s] wheel_scipy_like = PosixPath('/tmp/pytest-of-abuild/pytest-7/test0/mesonpy-test-n9jzgwby/mypkg-2.3.4-cp38-cp38-linux_x86_64.whl')
[   24s] 
[   24s]     @pytest.mark.skipif(win_py37, reason='An issue with missing file extension')
[   24s]     def test_scipy_like(wheel_scipy_like):
[   24s]         # This test is meant to exercise features commonly needed by a regular
[   24s]         # Python package for scientific computing or data science:
[   24s]         #   - C and Cython extensions,
[   24s]         #   - including generated code,
[   24s]         #   - using `install_subdir`,
[   24s]         #   - packaging data files with extensions not known to Meson
[   24s]         artifact = wheel.wheelfile.WheelFile(wheel_scipy_like)
[   24s]     
[   24s]         expecting = {
[   24s]             'mypkg-2.3.4.dist-info/METADATA',
[   24s]             'mypkg-2.3.4.dist-info/RECORD',
[   24s]             'mypkg-2.3.4.dist-info/WHEEL',
[   24s]             'mypkg/__init__.py',
[   24s]             'mypkg/__config__.py',
[   24s]             f'mypkg/extmod{EXT_SUFFIX}',
[   24s]             f'mypkg/cy_extmod{EXT_SUFFIX}',
[   24s]         }
[   24s]         # Meson master has a fix for `install_subdir` that is not present in
[   24s]         # 0.63.2: https://github.com/mesonbuild/meson/pull/10765
[   24s]         # A backport of the fix may land in 0.63.3, if so then remove the version
[   24s]         # check here and add the two expected files unconditionally.
[   24s]         if meson_version and meson_version >= '0.63.99':
[   24s]             expecting |= {
[   24s]                 'mypkg/submod/__init__.py',
[   24s]                 'mypkg/submod/unknown_filetype.npq',
[   24s]             }
[   24s]         if os.name == 'nt':
[   24s]             # Currently Meson is installing `.dll.a` (import libraries) next to
[   24s]             # `.pyd` extension modules. Those are very small, so it's not a major
[   24s]             # issue - just sloppy. For now, ensure we don't fail on those
[   24s]             actual_files = wheel_contents(artifact)
[   24s]             for item in expecting:
[   24s]                 assert item in actual_files
[   24s]         else:
[   24s] >           assert wheel_contents(artifact) == expecting
[   24s] E           AssertionError: assert {'mypkg-2.3.4.dist-info/METADATA',\n 'mypkg-2.3.4.dist-info/RECORD',\n 'mypkg-2.3.4.dist-info/WHEEL',\n 'mypkg/__config__.py',\n 'mypkg/__init__.py',\n 'mypkg/cy_extmod.cpython-38-x86_64-linux-gnu.so',\n 'mypkg/extmod.cpython-38-x86_64-linux-gnu.so',\n 'mypkg/submod/__init__.py',\n 'mypkg/submod/unknown_filetype.npq'} == {'mypkg-2.3.4.dist-info/METADATA',\n 'mypkg-2.3.4.dist-info/RECORD',\n 'mypkg-2.3.4.dist-info/WHEEL',\n 'mypkg/__config__.py',\n 'mypkg/__init__.py',\n 'mypkg/cy_extmod.cpython-38-x86_64-linux-gnu.so',\n 'mypkg/extmod.cpython-38-x86_64-linux-gnu.so'}
[   24s] E             Extra items in the left set:
[   24s] E             'mypkg/submod/__init__.py'
[   24s] E             'mypkg/submod/unknown_filetype.npq'
[   24s] E             Full diff:
[   24s] E               {
[   24s] E                'mypkg-2.3.4.dist-info/METADATA',
[   24s] E                'mypkg-2.3.4.dist-info/RECORD',
[   24s] E                'mypkg-2.3.4.dist-info/WHEEL',
[   24s] E                'mypkg/__config__.py',
[   24s] E                'mypkg/__init__.py',
[   24s] E                'mypkg/cy_extmod.cpython-38-x86_64-linux-gnu.so',
[   24s] E                'mypkg/extmod.cpython-38-x86_64-linux-gnu.so',
[   24s] E             +  'mypkg/submod/__init__.py',
[   24s] E             +  'mypkg/submod/unknown_filetype.npq',
[   24s] E               }
[   24s] 
[   24s] tests/test_wheel.py:123: AssertionError
[   24s] ---------------------------- Captured stdout setup -----------------------------
[   24s] + meson setup --native-file=/home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-native-file.ini -Ddebug=false -Doptimization=2 --prefix=/usr /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-3rrv641k/build
[   24s] The Meson build system
[   24s] Version: 0.63.3
[   24s] Source dir: /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like
[   24s] Build dir: /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-3rrv641k/build
[   24s] Build type: native build
[   24s] Project name: scipy-like
[   24s] Project version: 2.3.4
[   24s] C compiler for the host machine: cc (gcc 12.2.1 "cc (SUSE Linux) 12.2.1 20220830 [revision e927d1cf141f221c5a32574bde0913307e140984]")
[   24s] C linker for the host machine: cc ld.bfd 2.39.0.20220810-1
[   24s] Cython compiler for the host machine: cython (cython 0.29.32)
[   24s] Host machine cpu family: x86_64
[   24s] Host machine cpu: x86_64
[   24s] Program python found: YES (/usr/bin/python3.8)
[   24s] Found pkg-config: /usr/bin/pkg-config (1.8.0)
[   24s] Build targets in project: 3
[   24s] 
[   24s] scipy-like 2.3.4
[   24s] 
[   24s]   User defined options
[   24s]     Native files: /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-native-file.ini
[   24s]     debug       : false
[   24s]     optimization: 2
[   24s]     prefix      : /usr
[   24s] 
[   24s] Found ninja-1.11.1 at /usr/bin/ninja
[   24s] + meson compile
[   24s] [1/6] Generating mypkg/generate-config-file with a custom command
[   24s] [2/6] Compiling C object mypkg/extmod.cpython-38-x86_64-linux-gnu.so.p/extmod.c.o
[   24s] [3/6] Linking target mypkg/extmod.cpython-38-x86_64-linux-gnu.so
[   24s] [4/6] Compiling Cython source /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/mypkg/cy_extmod.pyx
[   24s] [5/6] Compiling C object mypkg/cy_extmod.cpython-38-x86_64-linux-gnu.so.p/meson-generated_mypkg_cy_extmod.pyx.c.o
[   24s] [6/6] Linking target mypkg/cy_extmod.cpython-38-x86_64-linux-gnu.so
[   24s] + meson install --destdir /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-3rrv641k/install
[   24s] ninja: Entering directory `/home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-3rrv641k/build'
[   24s] [1/1] Generating mypkg/generate-config-file with a custom command
[   24s] Installing subdir /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/mypkg/submod to /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-3rrv641k/install/usr/lib64/python3.8/site-packages/mypkg/submod
[   24s] Installing /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/mypkg/submod/__init__.py to /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-3rrv641k/install/usr/lib64/python3.8/site-packages/mypkg/submod
[   24s] Installing /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/mypkg/submod/unknown_filetype.npq to /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-3rrv641k/install/usr/lib64/python3.8/site-packages/mypkg/submod
[   24s] Installing mypkg/__config__.py to /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-3rrv641k/install/usr/lib64/python3.8/site-packages/mypkg
[   24s] Installing mypkg/extmod.cpython-38-x86_64-linux-gnu.so to /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-3rrv641k/install/usr/lib64/python3.8/site-packages/mypkg
[   24s] Installing mypkg/cy_extmod.cpython-38-x86_64-linux-gnu.so to /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-3rrv641k/install/usr/lib64/python3.8/site-packages/mypkg
[   24s] Installing /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/mypkg/__init__.py to /home/abuild/rpmbuild/BUILD/meson_python-0.10.0/tests/packages/scipy-like/.mesonpy-3rrv641k/install/usr/lib64/python3.8/site-packages/mypkg
[   24s] Copying files to wheel...
[   24s] [0/4] mypkg/__config__.py
[   24s] [1/4] mypkg/extmod.cpython-38-x86_64-linux-gnu.so
[   24s] [2/4] mypkg/cy_extmod.cpython-38-x86_64-linux-gnu.so
[   24s] [3/4] mypkg/__init__.py
[   24s] [4/4] mypkg/submod
[   24s] =========================== short test summary info ============================
[   24s] FAILED tests/test_wheel.py::test_scipy_like - AssertionError: assert {'mypkg-...
[   24s] =================== 1 failed, 51 passed, 1 xpassed in 18.96s ===================
```

Apparently https://github.com/mesonbuild/meson/pull/10765 has been backported to 0.63.3

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -2,7 +2,7 @@
 build-backend = 'mesonpy'
 backend-path = ['.']
 requires = [
-  'meson>=0.63.0',
+  'meson>=0.63.3',
   'ninja',
   'pyproject-metadata>=0.5.0',
   'tomli>=1.0.0',
@@ -26,7 +26,7 @@ classifiers = [
 
 dependencies = [
   'colorama; os_name == "nt"',
-  'meson>=0.62.0',
+  'meson>=0.63.3',
   'ninja',
   'pyproject-metadata>=0.5.0', # not a hard dependency, only needed for projects that use PEP 621 metadata
   'tomli>=1.0.0',

--- a/tests/test_wheel.py
+++ b/tests/test_wheel.py
@@ -97,16 +97,9 @@ def test_scipy_like(wheel_scipy_like):
         'mypkg/__config__.py',
         f'mypkg/extmod{EXT_SUFFIX}',
         f'mypkg/cy_extmod{EXT_SUFFIX}',
+        'mypkg/submod/__init__.py',
+        'mypkg/submod/unknown_filetype.npq',
     }
-    # Meson master has a fix for `install_subdir` that is not present in
-    # 0.63.2: https://github.com/mesonbuild/meson/pull/10765
-    # A backport of the fix may land in 0.63.3, if so then remove the version
-    # check here and add the two expected files unconditionally.
-    if meson_version >= '0.63.3':
-        expecting |= {
-            'mypkg/submod/__init__.py',
-            'mypkg/submod/unknown_filetype.npq',
-        }
     if os.name == 'nt':
         # Currently Meson is installing `.dll.a` (import libraries) next to
         # `.pyd` extension modules. Those are very small, so it's not a major

